fvm/state/address_generator_state.go

```diff
@@ -0,0 +1,67 @@
+package state
+
+import (
+	"github.com/onflow/flow-go/model/flow"
+)
+
+const keyAddressState = "account_address_state"
+
+type AddressGeneratorState struct {
+	ledger Ledger
+	chain  flow.Chain
+}
+
+func NewAddressGeneratorState(ledger Ledger, chain flow.Chain) *AddressGeneratorState {
+	return &AddressGeneratorState{
+		ledger: ledger,
+		chain:  chain,
+	}
+}
+
+func (state *AddressGeneratorState) Init() {
+	state.Set(state.chain.NewAddressGenerator())
+}
+
+func (state *AddressGeneratorState) GetGenerator() (flow.AddressGenerator, error) {
+	stateBytes, err := state.ledger.Get("", "", keyAddressState)
+	if err != nil {
+		return nil, err
+	}
+
+	addressGenerator := state.chain.BytesToAddressGenerator(stateBytes)
+
+	return &stateBoundAddressGenerator{
+		addressGenerator:      addressGenerator,
+		addressGeneratorState: state,
+	}, nil
+}
+
+func (state *AddressGeneratorState) Set(generator flow.AddressGenerator) {
```

Comment:
It looks like this can probably be unexported. What about these two functions for parity?

```go
func (state *AddressGeneratorState) GetGenerator() (flow.AddressGenerator, error) { ... }

func (state *AddressGeneratorState) getGeneratorState() ([]byte, error) { ... }
func (state *AddressGeneratorState) setGeneratorState(generator flow.AddressGenerator) { ... }